### PR TITLE
Feature: Toggle option(ignore_all_space, word_diff) for diff view

### LINKED ISSUE #173: Exclude whitepsaces from diff
It might be a nice addition to be able, if needed, to exclude whitespaces from diffs. Sometimes you are modifying someone elses' code and they don't follow the same standard and reviewing changes is a bit of pain if it has tons of unnecessary whitespaces changes (I have my editor to trim trailing whitespaces).

### PR DESCRIPTION
**Pull request by stoivo**
Closes #173.

### CODE REVIEW

**Comment by stoivo** [comment]
It is not ready to get merged. We need some syntax file for word-diff

File: core/commands/diff.py [hunk 1]
```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
     """
 
     def run(self, edit, cursors=None):
-        in_cached_mode = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
-        base_commit = self.view.settings().get("git_savvy.diff_view.base_commit")
+        in_cached_mode      = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
+        in_ignore_all_space = self.view.settings().get("git_savvy.diff_view.in_ignore_all_space")
+        in_word_diff        = self.view.settings().get("git_savvy.diff_view.in_word_diff")
+        base_commit         = self.view.settings().get("git_savvy.diff_view.base_commit")
```
Comment (divmain) [inline core/commands/diff.py]: Please no `=` alignment.  I stick mostly to PEP8, and this does not conform.

File: core/commands/diff.py [hunk 2]
```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
...
+            "--",
+            self.file_path)
 
         self.view.run_command("gs_replace_view_text", {"text": stdout})
 
+class GsDiffSetSettingAndRefresh(WindowCommand):
+
+    """
+    Set a setting to the diff view like: in_ignore_all_space, in_word_diff
+    Refresh the view after
```
Comment (divmain) [inline core/commands/diff.py]: Please edit description:
```
Toggle view settings.  Either `in_ignore_all_space` or `in_word_diff`.
```

File: core/commands/diff.py [hunk 3]
```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
...
+            "--ignore-all-space" if in_ignore_all_space else None,
+            "--word-diff" if in_word_diff else None,
+            "--no-color", base_commit,
+            "--cached" if in_cached_mode else None,
+            "--",
+            self.file_path)
 
         self.view.run_command("gs_replace_view_text", {"text": stdout})
 
+class GsDiffSetSettingAndRefresh(WindowCommand):
```
Comment (divmain) [inline core/commands/diff.py]: Change this to
```python
class GsDiffSetSettingAndRefresh(TextCommand):
```

File: core/commands/diff.py [hunk 4]
```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
...
+
+    """
+    Set a setting to the diff view like: in_ignore_all_space, in_word_diff
+    Refresh the view after
+    """
+
+    def run(self, setting):
+        setting_str = "git_savvy.diff_view.{}".format(setting)
+        print(setting_str)
+        view = self.window.active_view()
```
Comment (divmain) [inline core/commands/diff.py]: By changing this to `TextCommand` subclass, no need to check `self.window` for the active view.  Just use `self.view`.

File: Default.sublime-keymap [hunk 5]
```diff
@@ -346,6 +346,22 @@
             { "key": "setting.git_savvy.diff_view", "operator": "equal", "operand": true }
         ]
     },
+    {
+        "keys": ["s"],
+        "command": "gs_diff_set_setting_and_refresh",
```
Comment (divmain) [inline Default.sublime-keymap]:
```
"command": "gs_diff_toggle_setting"
```

File: core/commands/diff.py [hunk 6]
```diff
@@ -82,6 +107,12 @@ class GsDiffStageOrResetHunkCommand(TextCommand, GitCommand):
     """
 
     def run(self, edit, reset=False):
+        in_ignore_all_space = self.view.settings().get("git_savvy.diff_view.in_ignore_all_space")
+        in_word_diff        = self.view.settings().get("git_savvy.diff_view.in_word_diff")
+        if in_ignore_all_space or in_word_diff:
+            sublime.error_message("You have to be in a clean diff to stage.")
```
Comment (divmain) [inline core/commands/diff.py]: :+1:

**Comment by divmain** [comment]
This is great @stoivo - thanks!  I think we should leave the syntax highlighting until a later time - I don't think it'll be necessary to have this to start.

I had a handful of comments, and please fix the alignment issues (in case you're unfamiliar, you can read up on [PEP8 guidelines](https://www.python.org/dev/peps/pep-0008/) and search for `assignment`).

Once that is ready, I'll merge this in!

File: core/commands/diff.py [hunk 7]
```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
...
+class GsDiffSetSettingAndRefresh(WindowCommand):
+
+    """
+    Set a setting to the diff view like: in_ignore_all_space, in_word_diff
+    Refresh the view after
+    """
+
+    def run(self, setting):
+        setting_str = "git_savvy.diff_view.{}".format(setting)
+        print(setting_str)
```
Comment (stoivo) [inline core/commands/diff.py]: I don't remember why this is there. I will remove it :smiley: Probably from  debug.

Comment (divmain) [inline core/commands/diff.py]: I missed that - thanks!

**Comment by stoivo** [comment]
fixed. Ready to get merged

**Pull request merged**